src/interfaces/wallet.cpp

```diff
@@ -491,7 +491,7 @@ class WalletClientImpl : public ChainClient
     }
     bool verify() override { return VerifyWallets(m_chain, m_wallet_filenames); }
     bool load() override { return LoadWallets(m_chain, m_wallet_filenames); }
-    void start(CScheduler& scheduler) override { return StartWallets(scheduler); }
+    void start(CScheduler& scheduler, const ArgsManager& argsman) override { return StartWallets(scheduler, argsman); }
```

Comment:
In commit "wallet: Pass unused argsman to StartWallets()" (faccdaba583288273399481b1f7892a910080fda)

Not a big deal and I can fix it later, but this is a bit broken with #10102 because it is trying to pass a whole argsman object from the node to wallet process just to pass a single bool. (See "Interface method parameter and return types should either be serializable or be other interface classes" examples in https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#internal-interface-guidelines)

Recommended fix for this would be to pass `-flushwallet` bool setting to StartWallets the same way the `-wallet` list setting gets passed to LoadWallets above: first read in [`WalletInit::Construct`](https://github.com/bitcoin/bitcoin/blob/73407ff65cefa29ce522cb3cb43898497e8afa30/src/wallet/init.cpp#L120) and passed as an argument to `MakeWalletClient`, stored in the WalletClientImpl instance, and used as needed

Reply:
Thanks, tried to implement your suggestion in branch 9ddddd2d927b3a752e4698c4771fc68cdc2d9698

Reply:
> Thanks, tried to implement your suggestion in branch [9ddddd2](https://github.com/bitcoin/bitcoin/commit/9ddddd2d927b3a752e4698c4771fc68cdc2d9698)

Branch looks good. Current version is fine too in case you prefer that or the suggestion doesn't work.

 I was thinking maybe the best another approach after #19096 would be to add an ArgsManager pointer to the WalletContext struct, and initializing it from a new `MakeWalletClient` `ArgsManager&` argument so the wallet filenames and periodic flush arguments and temporary storage could be dropped

Reply:
> add an ArgsManager pointer to the WalletContext struct

thx, done